orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt

```diff
@@ -267,30 +269,59 @@ class SqlExecutionRepository(
 
   override fun delete(type: ExecutionType, id: String) {
     doForeignAware(DeleteInterlinkEvent(type, id)) {
-      execution, dslContext ->
-      val correlationField = if (type == PIPELINE) "pipeline_id" else "orchestration_id"
-      val (ulid, _) = mapLegacyId(jooq, type.tableName, id)
+      _, dslContext ->
+      val (ulid, _) = mapLegacyId(dslContext, type.tableName, id)
 
-      val correlationId = jooq.select(field("id")).from("correlation_ids")
-        .where(field(correlationField).eq(ulid))
-        .limit(1)
-        .fetchOne()
-        ?.into(String::class.java)
-      val stageIds = jooq.select(field("id")).from(type.stagesTableName)
-        .where(field("execution_id").eq(ulid))
-        .fetch()
-        ?.into(String::class.java)?.toTypedArray()
-
-      if (correlationId != null) {
-        dslContext.delete(table("correlation_ids")).where(field("id").eq(correlationId)).execute()
-      }
-      if (stageIds != null) {
-        dslContext.delete(type.stagesTableName).where(field("id").`in`(*stageIds)).execute()
-      }
-      dslContext.delete(type.tableName).where(field("id").eq(ulid)).execute()
+      deleteInternal(dslContext, type, listOf(ulid))
+    }
+  }
+
+  /**
+   * Deletes given executions
+   * NOTE: this method explicitly does not check if the execution is foreign or not.
+   */
+  override fun delete(type: ExecutionType, idsToDelete: List<String>) {
+    jooq.transactional { tx ->
+      deleteInternal(tx, type, idsToDelete)
     }
   }
 
+  private fun deleteInternal(dslContext: DSLContext, type: ExecutionType, idsToDelete: List<String>) {
+    val correlationField = if (type == PIPELINE) "pipeline_id" else "orchestration_id"
```

Comment:
Nit: I'd rather be defensive and not assume that there are only 2 possible values here:

```kotlin
val correlationField = when (type) {
    PIPELINE -> "pipeline_id"
    ORCHESTRATION -> "orchestration_id"
    else -> throw IllegalStateException("Unexpected field $type")
}
```